src/content/1.7/modules/concepts/doctrine/how-to-handle-multi-lang-doctrine-entitiy.md

```diff
@@ -0,0 +1,512 @@
+- [Doctrine Association mapping](https://www.doctrine-project.org/projects/doctrine-orm/en/2.7/reference/association-mapping.html)
+- [Mastering Doctrine Relationships in Symfony 3 (video tutorial)](https://symfonycasts.com/screencast/symfony3-doctrine-relations)
+{{% /notice %}}
+
+## Create a translated entity
+
+We will here display a more basic usage of Doctrine entity manager to create a translated entity:
+
+```php
+<?php
+use PrestaShop\Module\DemoDoctrine\Entity\Quote;
+use PrestaShop\Module\DemoDoctrine\Entity\QuoteLang;
+use PrestaShopBundle\Entity\Lang;
+
+// Getting the container depends if you are in a module or a controller but the rest of the code remains the same
+$container = $this->getContainer();
+
+// Get all languages vie the Lang repository
+$langRepository = $container->get('prestashop.core.admin.lang.repository');
+$languages = $this->langRepository->findAll();
+
+// The entity manager will allow us to persist the Doctrine entities
+$entityManager = $container->get('doctrine.orm.default_entity_manager');
+
+// This are some fixtures data
+$quotesData [
+    [
+        'author' => 'Pierre Augustin Caron de Beaumarchais',
+        'quotes' => [
+          'en' => 'Proving that I am right would be admitting that I could be wrong.',
+          'fr' => 'Prouver que j\'ai raison serait accorder que je puis avoir tort.',
+        ]
+    ],
+    [
+        'author' => 'Georges Bernanos',
+        'quotes' => [
+          'en' => 'There are no half-truths.',
+          'fr' => 'Il n\'y a pas de verités moyennes.',
+        ]
+    ],
+];
+
+foreach ($quotesData as $quoteData) {
+    $quote = new Quote();
+    $quote->setAuthor($quoteData['author']);
+
+    /** @var Lang $language */
+    foreach ($languages as $language) {
+        $quoteLang = new QuoteLang();
+        $quoteLang->setLang($language);
+        if (isset($quoteData['quotes'][$language->getIsoCode()])) {
+            $quoteLang->setContent($quoteData['quotes'][$language->getIsoCode()]);
+        } else {
+            $quoteLang->setContent($quoteData['quotes']['en']);
+        }
+        $quote->addQuoteLang($quoteLang);
+    }
+
+    // Usually we should also persist the QuoteLang entities, but since we setup the cascading persist in the Quote
+    // entity it is not necessary here, this action allows the entity manager to be aware of this new entity
+    $entityManager->persist($quote);
+}
+
+// Finally the entity manager applies ALL the creation in one go
+$entityManager->flush();
+```
+
+## Fetching your multi lang entity
+
+### Creating your repository service
+
+It is more efficient to have a dedicated repository service to handle your entity, as mentioned when creating the `Quote` entity a repository `PrestaShop\Module\DemoDoctrine\Repository\QuoteRepository`
+has been configured and assigned to this entity, this allows the Doctrine entity manager to map them, but we can also define our own service so that we can get it via a service name.
```

Comment:
```suggestion
It is more efficient to have a dedicated repository service to handle your entity, as mentioned when creating the `Quote` entity a repository `PrestaShop\Module\DemoDoctrine\Repository\QuoteRepository` has been configured and assigned to this entity, this allows the Doctrine entity manager to map them, but we can also define our own service so that we can get it via a service name.
```

Reply:
done